mods/chat.js

```diff
@@ -1,573 +1,603 @@
+                    }]
                 },
-                JSON.stringify ( { chat: chat, user: user, access: rank } )
-                );
+                {
+                    attr: [{
+                        name: 'onclick',
+                        value: 'idrinth.chat.useroptions(' + chat + ',' + user + ',\'User\');this.parentNode.parentNode.removeChild(this.parentNode);'
+                    }]
+                }
+            ];
+            promotionModes.unshift(popupContent[0]);
+            promotionModes.push(popupContent[1]);
+            popupContent = promotionModes;
+        }
+        for (var i = 0, l = popupContent.length; i < l; l++) {
+            var prom = popupContent[i],
+                baseType = {type: 'li', content: prom.title, attributes: prom.attr};
+            options.push(idrinth.ui.buildElement(baseType));
+        }
+        var list = document.createElement('ul');
+        for (var count = 0, len = options.length; count < len; count++) {
+            list.appendChild(options[count]);
+        }
+        list.setAttribute('class', 'idrinth-userinfo-box');
+        var pos = idrinth.ui.getPosition(element);
+        list.setAttribute('style', 'left:' + pos.x + 'px;top:' + pos.y + 'px');
+        idrinth.ui.body.appendChild(list);
+    },
+    useroptions: function (chat, user, rank) {
+        idrinth.runAjax(
+            'https://dotd.idrinth.de/' + idrinth.platform + '/chat-service/rank/',
+            function (reply) {
+                try {
+                    reply = JSON.parse(reply);
+                    idrinth.alert(reply.message);
+                } catch (e) {
+                    idrinth.log(e);
+                }
+            },
+            function (reply) {
+                idrinth.alert(this.getMsg('modify.fail'));
+            },
+            function (reply) {
+                idrinth.alert(this.getMsg('modify.fail'));
+            },
+            JSON.stringify({chat: chat, user: user, access: rank})
+        );
     },
-    replaceInText: function ( message, regex, callbacks, lastField ) {
-        var complexHandler = function ( message, regex, callbacks, lastField ) {
-            var partHandler = function ( count, callbacks, text, textcontent ) {
-                var callbackHandler = function ( textcontent, func, text ) {
-                    var tmp = func ( text );
+    replaceInText: function (message, regex, callbacks, lastField) {
+        var complexHandler = function (message, regex, callbacks, lastField) {
+            var partHandler = function (count, callbacks, text, textcontent) {
+                var callbackHandler = function (textcontent, func, text) {
+                    var tmp = func(text);
                     for (var c2 = 0; c2 < tmp.length; c2++) {
-                        if ( tmp[c2] !== undefined ) {
-                            textcontent.push ( tmp[c2] );
+                        if (tmp[c2] === undefined) {
+                            continue;

```

Comment:
continue should be replaced by a function with early return statement or a wrapping condition